Nonprofit-Exchange-Hub/web-app
# Integrate create org

## Dev Summary
- Created types for form data
- Added `React Hook Form` and `Yup` for form logic and form validation to `client/package*`
- Added `react-query` and `axios` to deal with multiple interdependent api requests and errors
- Removed manual validation methods and replaced with `Yup`
- Expanded `TextField` wrapper component props
- Added logic for error messages


## Test Plan
repro steps:
Enter info for non-existing orgs or users
1. go to http://localhost:3000/signup-nonprofit
2. Fill ein in  step 1
3. Fill out org info in  step 2
4. Fill out user info in step 3
5. click submit
6. expect to see a success notification
7. Click submit again and expect to see a notification that this user already exists


#### Expect
[See my video demo](https://www.loom.com/share/a96766ed4f5f407e96f65f834c170686)


## Resources

- [simple validation](https://thewebdev.info/2022/01/06/how-to-invalidate-a-textfield-in-react-material-ui/)
- [formik/gulp ways](https://formik.org/docs/examples/with-material-ui)
- [Formik docs](https://formik.org/docs/overview)
- [Yup Docs](https://www.npmjs.com/package/yup)
